Sources/FoundationInternationalization/TimeZone/TimeZone_ObjC.swift

```diff
@@ -212,74 +227,62 @@ final class _NSSwiftTimeZone: _NSTimeZoneBridge {
     override func localizedName(_ style: TimeZone.NameStyle, locale: Locale?) -> String? {
         timeZone.localizedName(for: style, locale: locale)
     }
-}
-
-// MARK: -
-
-/// Wraps an `NSTimeZone` with a more Swift-like `TimeZone` API.
-/// This is only used in the case where we have a custom Objective-C subclass of `NSTimeZone`.
-internal final class _NSTimeZoneSwiftWrapper: @unchecked Sendable {
-    let _timeZone: NSTimeZone
-
-    // MARK: -
-    // MARK: Bridging
-
-    internal init(adoptingReference reference: NSTimeZone) {
-        _timeZone = reference
-    }
-
-    func bridgeToObjectiveC() -> NSTimeZone {
-        return _timeZone.copy() as! NSTimeZone
-    }
-
-    func hash(into hasher: inout Hasher) {
-        hasher.combine(_timeZone)
-    }
+    
+    private static func dataFromTZFile(_ name: String) -> Data {
+        let path = TimeZone.TZDIR + "/" + name
+        guard !path.contains("..") else {
+            // No good reason for .. to be present anywhere in the path
+            return Data()
+        }
 
-    func isEqual(to other: Any) -> Bool {
-        if let other = other as? _NSTimeZoneSwiftWrapper {
-            return _timeZone == other._timeZone
-        } else if let other = other as? _TimeZone {
-            return self.identifier == other.identifier && self.data == other.data
-        } else {
-            return false
+        #if os(Windows)
+        let fd: CInt = path.withCString(encodedAs: UTF16.self) {
+            var fd: CInt = -1
+            let errno: errno_t =
+                _wsopen_s(&fd, $0, _O_RDONLY | _O_BINARY, _SH_DENYNO, _S_IREAD | _S_IWRITE)
+            guard errno == 0 else { return -1 }
+            return fd
         }
-    }
+        #else
+        let fd = open(path, O_RDONLY, 0666)
```

Comment:
Looking at what we have now, we _already_ have wrappers for the `open()` functions. We should probably fix the Windows ones to do a proper path conversion, and it might make sense to declare `O_RDONLY` et al without underscores for Windows too.

Reply:
It's actually worse than that - we don't have general agreement on the normalized UTF8 form (NFC, NFD). It's likely we're doing it wrong by using a String as a Path in the vast majority of cases. That problem is too big to solve here, though (and anyway, this code just moved).

Reply:
Agreed, let's not change this for now, but we should aim to do _something_ about it.